ccpage131.py

amymhaddad opened the pull request:
There are several short exercises listed in this file. The instructions for each exercise are listed as comments. I intentionally use a double space to separate the exercises.

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+sandwich_orders = ['tuna', 'beef', 'peanut butter', 'ham']
+
+# use while loop to loop through the orders in sandwich_orders
+while sandwich_orders:
```
Awesome all around.

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+while sandwich_orders:
+    completed_sandwich = sandwich_orders.pop()
+    finished_sandwiches.append(completed_sandwich)
+    
```
:(

amymhaddad replied:
@paulghaddad Shouldn't there be a whitespace here? This print statement goes along with the for loop that follows and is separate from the while loop before it.

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+finished_sandwiches = []
+
+# Step 2: remove all instances of pastrami
+while 'pastrami' in sandwich_orders:
```
👏

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+while 'pastrami' in sandwich_orders:
+    sandwich_orders.remove('pastrami')
+
+while sandwich_orders:
```
Wow, super impressed!

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+
+
+# Write a program that polls users about their dream vacation.
+# Write a prompt similar to If you could visit one place in the world, where would you go? 
```
:(

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+    name = input("What's your name? ")
+    dream_vacation_location = input("Where do you want to travel to? ")
+
+    responses[name] = dream_vacation_location
```
Great job!

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+    responses[name] = dream_vacation_location
+
+    print("Is there another person who should take this poll?")
+    repeat_with_new_user = input("Type: 'yes' or 'no': ")
```
Perhaps a better variable name is `user_input`?

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+
+    print("Is there another person who should take this poll?")
+    repeat_with_new_user = input("Type: 'yes' or 'no': ")
+    normalized_repeat_with_new_user = repeat_with_new_user.lower()
```
Nice awareness of normalizing all user input.

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+    normalized_repeat_with_new_user = repeat_with_new_user.lower()
+
+    if normalized_repeat_with_new_user == 'no':
+        active = False
```
An alternative is to use `break` (which would require you to adjust the loop to `while True:`.

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+
+    if normalized_repeat_with_new_user == 'no':
+        active = False
+    
```
:(

amymhaddad replied:
@paulghaddad This line break is intentional to break up the different exercises, since this file contains a bunch of exercises and not one program.

paulghaddad commented on ccpage131.py:
```diff
@@ -0,0 +1,77 @@
+# Write loop to print results
+print("These are the results of the poll: ")
+for name, location in responses.items():
+    print(f"\nName: {name.capitalize()}")
```
Recommend moving the line break to the end of the second `print` statement.

amymhaddad commented:
@paulghaddad I moved the line break on line 74, removed the trailing whitespaces, and changed the variable name on line 67. I also responded back to several of your comments.
